src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -973,22 +973,33 @@ public function getDirty()
     }
 
     /**
-     * Determine if the new and old values for a given key are numerically equivalent.
+     * Determine if the new and old values for a given key are equivalent.
      *
      * @param  string  $key
      * @return bool
      */
-    protected function originalIsNumericallyEquivalent($key)
```

Comment:
Now I got your point, we can leave the `originalIsNumericallyEquivalent` in the code and add the `originalIsEquivalent`